### PR TITLE
Update plot_series to handle pd.Int64 and pd.Range index uniformly

### LINKED ISSUE #882: The function 'plot_series' should accept series with different index types
**Is your feature request related to a problem? Please describe.**
The plot_series function throws an error when multiple series with different index types are passed as parameters.

**Describe the solution you'd like**
The function should include implicit type casting of the series' indexes  

**Describe alternatives you've considered**
Manually converting the datatype of the series' indexes before passing them to the plot_series function using pandas 

**Additional context**
Screenshot 2021-05-20 at 4 48 00 PM

### PR DESCRIPTION
**Pull request by Dbhasin1**
#### Reference Issues/PRs
Fixes #882 

#### What does this implement/fix? Explain your changes.
added the provision of typecasting different types of series indexes such as Int64, Float64, UInt64 and RangeIndex into each other 

#### Does your contribution introduce a new dependency? If yes, which one?
No

### CODE REVIEW

File: sktime/utils/plotting.py
Comment (mloning) [inline sktime/utils/plotting.py:74]: Note that we already check if the index is valid above in `check_y(y)`.

File: sktime/utils/plotting.py
Comment (mloning) [inline sktime/utils/plotting.py:75]: I would try something like this: 
```python
def _is_int_index(index):
    return type(index) in (pd.Int64Index, pd.RangeIndex)


def _check_consistent_index_type(*series):
    
    index = series[0].index    
    for y in series[1:]: 
        
        msg = "Found `series` with inconsistent index types. Please make sure that all series have the same index type."
        
        if _is_int_index(y.index):
            if not _is_int_index(index):
                raise TypeError(msg)
                
        else:
            if type(index) is not type(y.index):
                raise TypeError(msg)
```

We can then use the function `_check_consistent_index_type` at the beginning of the `plot_series` function. What do you think?

**Review by mloning — COMMENTED** [review]
Thanks @Dbhasin1 for the PR and participation today in the summer programme (#863)!

File: sktime/utils/plotting.py
Comment (Dbhasin1) [reply sktime/utils/plotting.py:74]: Yep, I just went through the code for this function and it seems like the only accepted index types as of now are Int64Index, RangeIndex, PeriodIndex and pd.DatetimeIndex. Can we add Float64Index and UInt64Index in this list too?

File: sktime/utils/plotting.py
Comment (Dbhasin1) [reply sktime/utils/plotting.py:75]: Yes, this makes complete sense. I realised that the 'union' method includes implicit type casting, so there should be no need for manually doing so in the code now.

File: sktime/utils/plotting.py
Comment (mloning) [reply sktime/utils/plotting.py:74]: Float64Index and UInt64Index index types don't really make sense for time indices I think, at least not for time series with equidistant time points. Why would you want to add them?

Comment (Dbhasin1) [reply sktime/utils/plotting.py:74]: I was thinking more along the lines of how the function shouldn't throw an error if the user passes a series with index values as '1.0, 2.0, 3.0, and so on.' But yeah, otherwise, including them doesn't really make sense, so we could leave them off

Comment (mloning) [reply sktime/utils/plotting.py:74]: Ah okay, I see, but I think it's probably safer to tell people to use an integer index in that case rather than trying to convert a float index.

File: sktime/utils/validation/series.py
Comment (mloning) [inline sktime/utils/validation/series.py:165]: Let's make this more explicit here, because the order of `VALID_INDEX_TYPES` may change

File: sktime/utils/validation/series.py
Comment (mloning) [inline sktime/utils/validation/series.py:201]: I wouldn't include this operation into this function, it's not directly related to the checking of consistent index types.

File: sktime/utils/plotting.py
Comment (mloning) [inline sktime/utils/plotting.py:5]:
```suggestion
"""Functionality for plotting one or more time series."""
```

File: sktime/utils/plotting.py
Comment (mloning) [inline sktime/utils/plotting.py:13]: This should be above the imports.

File: sktime/utils/plotting.py
Comment (mloning) [inline sktime/utils/plotting.py:28]:
```suggestion
        The length of the list has to match with the number of series. 
```

File: sktime/utils/validation/series.py
Comment (mloning) [inline sktime/utils/validation/series.py:181]: could we move this comment to line 189 below

File: sktime/utils/validation/series.py
Comment (mloning) [inline sktime/utils/validation/series.py:169]:
```suggestion
    """Check that index types are consistent with each other.
```

**Review by mloning — COMMENTED** [review]
Great @Dbhasin1, the doc tests are passing now, just a few more minor comments and we're good to go after that I think!

**Comment by Dbhasin1** [comment]
> Great @Dbhasin1, the doc tests are passing now, just a few more minor comments and we're good to go after that I think!

That's awesome, I'll incorporate your suggested changes asap

**Comment by Dbhasin1** [comment]
hey, @mloning I don't know why the build (macOS-latest, 3.6) check got canceled. Could you give any pointers on how I could resolve this?

**Review by mloning — COMMENTED** [review]
Great @Dbhasin1 - looks all good now!

The last thing I would add is a unit test to make sure we address the issue in https://github.com/alan-turing-institute/sktime/issues/882. I would just use the code snippet in the issue and see if it now passes without any errors. Please also take a look at the time index (x-axis) and check that it makes sense.

**Review by mloning — DISMISSED** [review]
Hi @Dbhasin1, I've added the unit test and am happy to merge this now. Let me know if you have any final comments before merging?

**Review by mloning — APPROVED** [review]

**Comment by Dbhasin1** [comment]
> Hi @Dbhasin1, I've added the unit test and am happy to merge this now. Let me know if you have any final comments before merging?

Hi @mloning, sorry I couldn't add the unit test in time, got a little engaged elsewhere. yeah it looks good to me, may I know what you did to check whether the time index made sense?

**Comment by mloning** [comment]
Hi @Dbhasin1 take a look at the file changes here: https://github.com/alan-turing-institute/sktime/pull/892/files

**Comment by mloning** [comment]
Thanks again @Dbhasin1 for the PR, great work! As said, please reach out again if you want to join the mentoring scheme or have any questions.

**Comment by Dbhasin1** [comment]
> Thanks again @Dbhasin1 for the PR, great work! As said, please reach out again if you want to join the mentoring scheme or have any questions.

Thank you @mloning! Yep, will definitely do so!

**Pull request merged**
